src/thor/route_action.cc

```diff
@@ -425,11 +444,16 @@ void thor_worker_t::path_arrive_by(Api& api, const std::string& costing) {
           route = api.mutable_trip()->mutable_routes()->Add();
         auto& leg = *route->mutable_legs()->Add();
         TripLegBuilder::Build(api.options(), controller, *reader, mode_costing, path.begin(),
-                              path.end(), *origin, *destination, throughs, leg, interrupt, &vias);
+                              path.end(), *origin, *destination, throughs, leg, algorithms, interrupt,
+                              &vias);
         path.clear();
         vias.clear();
       }
     }
+
+    // if we just made a leg that means we are done recording which algorithms were used
+    if (path.empty())
```

Comment:
this is a bit tricky because of alternates. we have to wait until that loop is finished above before we clear this or the first alternate wont have algorithms set